CMakeLists.txt

```diff
@@ -201,6 +201,12 @@ if(CMAKE_COMPILER_IS_CLANG)
   SET(CLANG_LIBRARIES "stdc++")
 endif()
 
+if(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
+  if("${CMAKE_CXX_FLAGS}" STREQUAL "")
+    SET(CMAKE_CXX_FLAGS "${SSE_FLAGS_STR}")
```

Comment:
Shouldn't we be populating the compiler options instead. I'm just talking semantics here.

Reply:
Yes, long-term these should be public compile options assuming you want consuming projects to inherit them (which we should).  I just mirrored what the other compilers do here, but since this one is going to be a longer-term effort definitely yes.

I think a more explicit option `PCL_SSE_FLAGS` or `PCL_VECTORIZATION_FLAGS` or `PCL_PLATFORM_FLAGS` or something should be added (cached empty string variable, not an `option` call).  Usage would be something like `cmake .. -DPCL_SSE_FLAGS="-march=native"` or `cmake .. -DPCL_SSE_FLAGS="-xavx512"` or whatever, probably allow it to be a list.  Then instead of guarding against `"${CMAKE_CXX_FLAGS}" STREQUAL ""`, we would do something like

```cmake
if (PCL_SSE_FLAGS)
  target_compile_options(pcl PUBLIC ${PCL_SSE_FLAGS})
else()
  include(pcl_sse_flags) # I'll fix up the CMAKE_MODULE_PATH to allow this include
  pcl_check_for_sse()
  if (PCL_SSE_FLAGS)
    target_compile_options(pcl PUBLIC ${PCL_SSE_FLAGS})
  endif()
endif()
```

Realistically, creating a `pcl-interface` target is the right thing to do here -- all libs link against that, so we actually would do `target_compile_options(pcl-interface INTERFACE ${PCL_SSE_FLAGS})`.  No need to special-case on compilers.  Working with lists is preferred, no need to convert to strings.  When you do `target_link_libraries(pcl PUBLIC pcl-interface)`, the `INTERFACE_*_` properties propagate from `pcl-interface` to `pcl` :)